common/lib/xmodule/xmodule/course_module.py

```diff
@@ -32,6 +32,42 @@
 CATALOG_VISIBILITY_NONE = "none"
 
 
+def _extract_course_summary(course):
+    """
+    Extract course information from the course block for split.
+    """
+    return {
+        field: course.fields[field]
+        for field in CourseSummary.course_info_fields
+        if field in course.fields
+    }
+
+
+def _extract_course_summary_old_mongo(course):
```

Comment:
It's unfortunate that this modulestore detail has leaked up to this level. Would it be possible to keep the `extract_course_summary` call in each modulestore itself? And just call `modulestore.extract_course_summary(course)` below in `get_course_summaries`?